rs/registry/canister/src/mutations/node_management/do_replace_operator.rs

```diff
@@ -0,0 +1,829 @@
+
+        registry.apply_mutations_for_test(vec![
+            dc_mutation("dc1"),
+            operator_mutation(operator(1), provider(1), 10, "dc1"),
+            operator_mutation(operator(2), provider(1), 10, "dc1"),
+            operator_mutation(operator(3), provider(1), 10, "dc1"),
+            node_mutation(node(1), operator(1)),
+            node_mutation(node(2), operator(3)),
+        ]);
+
+        registry
+            .do_replace_operator_(
+                payload(operator(1), operator(2), &[node(1), node(2)]),
+                caller(1),
+            )
+            .assert_err_contains(&format!(
+                "Node {} does not belong to node operator {}",
+                node(2),
+                operator(1)
+            ));
+    }
+
+    #[test]
+    fn insufficient_node_allowance() {
+        let mut registry = Registry::new();
+
+        registry.apply_mutations_for_test(vec![
+            dc_mutation("dc1"),
+            operator_mutation(operator(1), provider(1), 10, "dc1"),
+            operator_mutation(operator(2), provider(1), 2, "dc1"),
+            node_mutation(node(1), operator(1)),
+            node_mutation(node(2), operator(1)),
+            node_mutation(node(3), operator(1)),
+        ]);
+
+        registry
+            .do_replace_operator_(
+                payload(operator(1), operator(2), &[node(1), node(2), node(3)]),
+                caller(1),
+            )
+            .assert_err_contains(&format!("Adding 3 nodes would overflow node allowance for node operator {} who has 2 remaining", operator(2)));
+    }
+
+    #[test]
+    fn update_all_records_correctly() {
+        let mut registry = Registry::new();
+
+        let first_node = generate_valid_node_id();
+        let second_node = generate_valid_node_id();
+        let third_node = generate_valid_node_id();
+        registry.apply_mutations_for_test(
+            vec![
+                dc_mutation("dc1"),
+                operator_mutation(operator(1), provider(1), 10, "dc1"),
+                operator_mutation(operator(2), provider(1), 10, "dc1"),
+                node_mutation(first_node.node_id(), operator(1)),
+                node_mutation(second_node.node_id(), operator(1)),
+                node_mutation(third_node.node_id(), operator(1)),
+            ]
+            .into_iter()
+            // These are the mutations required to have a compliant registry
+            .chain(get_mutations_to_achieve_invariancy(&[
+                &first_node,
+                &second_node,
+                &third_node,
+            ]))
+            .collect(),
+        );
+
+        let version_before = registry.latest_version();
+        registry
+            .do_replace_operator_(
+                payload(
+                    operator(1),
+                    operator(2),
+                    &[
+                        first_node.node_id(),
+                        second_node.node_id(),
+                        third_node.node_id(),
```

Comment:
Maybe do not include this so that we can verify that unspecified nodes are not touched.